gobblin-service/src/main/java/org/apache/gobblin/service/modules/orchestration/proc/LaunchDagProc.java

```diff
@@ -102,6 +103,8 @@ private void submitNextNodes(DagManagementStateStore dagManagementStateStore, Da
        DagProcUtils.submitJobToExecutor(dagManagementStateStore, dagNode, getDagId());
        log.info("Submitted job {} for dagId {}", DagManagerUtils.getJobName(dagNode), getDagId());
      }
+
+     DagProcUtils.sendEnforceFlowFinishDeadlineDagAction(dagManagementStateStore, getDagTask().getDagAction());
```

Comment:
rather than situating here in `submitNextNodes`, this might better belong within `act()`... and shouldn't it come after this line (in `act()`):
```
//Checkpoint the dag state, it should have an updated value of dag nodes
dagManagementStateStore.checkpointDag(dag.get());
```
?